### ISSUE TITLE
Incorrect "no such field" error

### ISSUE THREAD

**Issue by SamRodri** [rust-analyzer/rust-analyzer#5147]
In the latest version (v0.2.224) this code causes the incorrect error:

```rust
macro_rules! Type {
    () => { u32 };
}

struct Foo {
    bar: Type![],
}
impl Foo {
    fn new() -> Self {
        Foo {
            bar: 0 // <- incorrect error here
        }
    }
}
```

I first encountered this using the syn [Token!](https://docs.rs/syn/1.0.33/syn/macro.Token.html) macro.

**Comment by flodiebold** [comment]
We don't expand type-position macros yet.

**Comment by SamRodri** [comment]
It used to work, just with an unknown type, now the field is missing.
